web3/contract.py

```diff
@@ -926,157 +1385,62 @@ def call(
         :return: ``Caller`` object that has contract public functions
             and variables exposed as Python methods
         """
-        if transaction is None:
-            call_transaction: TxParams = {}
-        else:
-            call_transaction = cast(TxParams, dict(**transaction))
-
-        if 'data' in call_transaction:
-            raise ValueError("Cannot set 'data' field in call transaction")
-
-        if self.address:
-            call_transaction.setdefault('to', self.address)
-        if self.w3.eth.default_account is not empty:
-            # type ignored b/c check prevents an empty default_account
-            call_transaction.setdefault('from', self.w3.eth.default_account)  # type: ignore
-
-        if 'to' not in call_transaction:
-            if isinstance(self, type):
-                raise ValueError(
-                    "When using `Contract.[methodtype].[method].call()` from"
-                    " a contract factory you "
-                    "must provide a `to` address with the transaction"
-                )
-            else:
-                raise ValueError(
-                    "Please ensure that this contract instance has an address."
-                )
+        call_transaction = self._get_call_txparams(transaction)
 
-        block_id = parse_block_identifier(self.w3, block_identifier)
+        block_id = await async_parse_block_identifier(self.w3, block_identifier)
 
-        return call_contract_function(
+        return await async_call_contract_function(
             self.w3,
             self.address,
             self._return_data_normalizers,
             self.function_identifier,
             call_transaction,
-            block_id,
+            block_id,  # type: ignore
```

Comment:
Do you mind adding a comment here so we know why we're ignoring?

Reply:
This is because `BlockIdentifier` doesn't have an Awaitable type in types.py. I was thinking it was more of a wholistic question if/when the Awaitables should be created than creating it with this contract work

comment added in #2466